src/auto/dbus_connection.rs

```diff
@@ -0,0 +1,1264 @@
+        unsafe {
+            gio_sys::g_dbus_connection_set_exit_on_close(
+                self.to_glib_none().0,
+                exit_on_close.to_glib(),
+            );
+        }
+    }
+
+    pub fn signal_subscribe<
+        P: Fn(&DBusConnection, &str, &str, &str, &str, &glib::Variant) + 'static,
+    >(
+        &self,
+        sender: Option<&str>,
+        interface_name: Option<&str>,
+        member: Option<&str>,
+        object_path: Option<&str>,
+        arg0: Option<&str>,
+        flags: DBusSignalFlags,
+        callback: P,
+    ) -> u32 {
+        let callback_data: Box_<P> = Box_::new(callback);
+        unsafe extern "C" fn callback_func<
+            P: Fn(&DBusConnection, &str, &str, &str, &str, &glib::Variant) + 'static,
+        >(
+            connection: *mut gio_sys::GDBusConnection,
+            sender_name: *const libc::c_char,
+            object_path: *const libc::c_char,
+            interface_name: *const libc::c_char,
+            signal_name: *const libc::c_char,
+            parameters: *mut glib_sys::GVariant,
+            user_data: glib_sys::gpointer,
+        ) {
+            let connection = from_glib_borrow(connection);
+            let sender_name: Borrowed<GString> = from_glib_borrow(sender_name);
+            let object_path: Borrowed<GString> = from_glib_borrow(object_path);
+            let interface_name: Borrowed<GString> = from_glib_borrow(interface_name);
+            let signal_name: Borrowed<GString> = from_glib_borrow(signal_name);
+            let parameters = from_glib_borrow(parameters);
+            let callback: &P = &*(user_data as *mut _);
+            (*callback)(
+                &connection,
+                sender_name.as_str(),
+                object_path.as_str(),
+                interface_name.as_str(),
+                signal_name.as_str(),
+                &parameters,
+            );
+        }
+        let callback = Some(callback_func::<P> as _);
+        unsafe extern "C" fn user_data_free_func_func<
+            P: Fn(&DBusConnection, &str, &str, &str, &str, &glib::Variant) + 'static,
+        >(
+            data: glib_sys::gpointer,
+        ) {
+            let _callback: Box_<P> = Box_::from_raw(data as *mut _);
+        }
+        let destroy_call9 = Some(user_data_free_func_func::<P> as _);
+        let super_callback0: Box_<P> = callback_data;
+        unsafe {
+            gio_sys::g_dbus_connection_signal_subscribe(
+                self.to_glib_none().0,
+                sender.to_glib_none().0,
+                interface_name.to_glib_none().0,
+                member.to_glib_none().0,
+                object_path.to_glib_none().0,
+                arg0.to_glib_none().0,
+                flags.to_glib(),
+                callback,
+                Box_::into_raw(super_callback0) as *mut _,
+                destroy_call9,
+            )
+        }
+    }
+
+    pub fn signal_unsubscribe(&self, subscription_id: u32) {
```

Comment:
This should get a newtype wrapper, `struct SignalSubscriptionId(u32)`. Similar to `Object::connect()` etc